packages/tooling/src/lib/parameters-to-json-schema.js

```diff
@@ -20,83 +27,45 @@ function getBodyParam(pathOperation, oas) {
   if (!schema) return null;
 
   const cleanupSchemaDefaults = obj => {
+    // Run through the constructed JSON Schema and repair a few OpenAPI-specific oddities we might be experiencing.
     Object.keys(obj).forEach(prop => {
       if (obj[prop] === null) {
         // If the item is null, just carry on. Why do this in addition to `typeof obj[prop] == object`? Because
         // `typeof null` equates to `object` for "legacy reasons" apparently.
         // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null
       } else if (typeof obj[prop] === 'object') {
         cleanupSchemaDefaults(obj[prop]);
-      } else {
-        switch (prop) {
-          case 'additionalProperties':
-            // If it's set to `false`, don't bother adding it.
-            if (obj[prop] === false) {
-              delete obj[prop];
-            }
-            break;
-
-          case 'default':
-            if ('allowEmptyValue' in obj && obj.allowEmptyValue && obj[prop] === '') {
-              // If we have `allowEmptyValue` present, and the default is actually an empty string, let it through as
-              // it's allowed.
-            } else if (obj[prop] === '') {
-              delete obj[prop];
-            }
-            break;
-
-          case 'maxLength':
-            obj.maximum = obj[prop];
-            delete obj[prop];
-            break;
-
-          case 'minLength':
-            obj.minimum = obj[prop];
-            delete obj[prop];
-            break;
-
-          case 'type':
-            if (obj.type === 'array') {
-              if (!('items' in obj)) {
-                if ('properties' in obj) {
-                  // This is a fix to handle cases where someone may have typod `items` as `properties` on an array.
-                  // Since throwing a complete failure isn't ideal, we can see that they meant for the type to be
-                  // `object`, so we  can do our best to shape the data into what they were intendint it to be.
-                  // README-6R
```

Comment:
Since `@openapi-contrib/openapi-schema-to-json-schema` ignores `properties` if the schema type is an array, I'm opting to no longer support this mangled OAS usecase. Instead, if we have a document with `type: array`, and it has `properties`, `@openapi-contrib/openapi-schema-to-json-schema` will return it as `{ type: array }` and then we'll add `item: {}` to fill it out as a mixed-content input.